### PR TITLE
Add setting to allow double clicking file to open diff

### PR DESCRIPTION
**Pull request by ianhi**
closes: https://github.com/jupyterlab/jupyterlab-git/issues/616

- Added a setting `doubleClickDiff` (default false) that controls if doubling clicking a file in the changes tab opens the file or opens a diff of the file
    - I also added this setting to the git menu next to the sample staging check box
- Created an `open file`  action button and icon to replace the diff action button when doubling clicking opens a diff

This gif appears low quality when I view in the PR, when I click on it it opens in a new tab in higher quality:
double_click

Somewhat difficult to see the new action button due to gif quality, heres a screenshot of it:
image

### CODE REVIEW

File: schema/plugin.json [hunk 1]
```diff
@@ -17,6 +17,12 @@
       "description": "Disable all branch operations (new, switch) when there are changed/staged files",
       "default": false
     },
+    "doubleClickDiff": {
+      "type": "boolean",
+      "title": "Show diff on double click",
+      "description": "If true doubling clicking a file in the changed files list will open a diff",
```
Comment (kgryte) [inline schema/plugin.json:23]:
```suggestion
      "description": "If true, doubling clicking a file in the list of changed files will open a diff",
```

File: style/images/open-file.svg [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 15 15" fill="none" />
```
Comment (kgryte) [inline style/images/open-file.svg:1]:
```suggestion
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 15 15" fill="none">
```

Comment (ianhi) [reply style/images/open-file.svg]: yikes, good catch. that's what I get for cleaning up the file after I made the GIF. I  was also missing a `/>` in one of the paths.

**Comment by ianhi** [comment]
I applied the suggestions and also made the default setting `false` so as to not to change the default behavior.

File: src/components/FileList.tsx [hunk 3]
```diff
@@ -536,81 +633,45 @@ export class FileList extends React.Component<IFileListProps, IFileListState> {
...
+        this.props.model,
+        {
+          previousRef: { gitRef: 'HEAD' },
+          currentRef: { specialRef: currentRef }
+        },
+        this.props.renderMime,
+        !file.is_binary
+      );
+    } catch (reason) {
+      console.error(`Fail to open diff view for ${file.to}.\n${reason}`);
```
Comment (ianhi) [inline src/components/FileList.tsx:674]:
```suggestion
      console.error(`Failed to open diff view for ${file.to}.\n${reason}`);
```

**Comment by fcollonval** [comment]
Thanks for the PR @ianhi  (and for the latest grammar corrections) - I went a bit further by displaying the icon for opening the file at all time. As even the default behavior is not trivially known by users.

Another very nice PR to be merged once the tests passes.

**Comment by fcollonval** [comment]
@meeseeksdev backport to 0.11.x

**Pull request merged**